### PR TITLE
New "Random Fact" button on landing page

### PR DESCRIPTION
**Pull request by jamari-morrison**
Added a new "Random fact" button onto the landing page. It uses a java servlet that sends a json (made using gson) to a client side script. The client side scrip selects a random message from the json given by the server and displays it on the page.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/RandomMessageServlet.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Handles requests sent to the /hello URL. Try running a server and navigating to /hello! */
```
Comment (raymondji) [inline portfolio/src/main/java/com/google/sps/servlets/RandomMessageServlet.java]: Outdated comment?

File: portfolio/src/main/java/com/google/sps/servlets/RandomMessageServlet.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> possibleMessages = new ArrayList<>();
+    Gson gson = new Gson();
+    possibleMessages.add("I onced flipped a bottle on top of another bottle");
+    possibleMessages.add("The best Java IDE is IntelliJ. Eclipse isn't even an option");
+    possibleMessages.add("This button took me a little over 5 hours total to debug");
+    possibleMessages.add("My favorite manga is Berserk");
+    possibleMessages.add("My favorite anime is either your lie in april or attack on titan");
```
Comment (raymondji) [inline portfolio/src/main/java/com/google/sps/servlets/RandomMessageServlet.java:27]: image

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -52,6 +57,11 @@ <h2 class="wrapper">
     <!-- Note: each list item is a floating box in the background -->
     <div class="animation-area">
         <div class="title">Hi, I'm Jamari.</div>
+        <div  class="randomMessageDiv">
+            <p id="random-message-txt" class="randomMessageTxt">Click the button to get a fact about me!</p>
```
Comment (raymondji) [inline portfolio/src/main/webapp/index.html]: nit: formatting :)

**Review by raymondji — APPROVED** [review]
lgtm (looks good to me)! :)

The PR title could be a bit more general; it doesn't only show the button, but also shows the message

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (jamari-morrison) [reply portfolio/src/main/webapp/index.html]: oh yeah I forgot that cloud shell handles formatting really nicely for us with just one button :o

File: portfolio/src/main/java/com/google/sps/servlets/RandomMessageServlet.java (hunk 2 above)
Comment (jamari-morrison) [reply portfolio/src/main/java/com/google/sps/servlets/RandomMessageServlet.java:27]: image

File: portfolio/src/main/java/com/google/sps/servlets/RandomMessageServlet.java (hunk 1 above)
Comment (jamari-morrison) [reply portfolio/src/main/java/com/google/sps/servlets/RandomMessageServlet.java]: Yeah, good catch!

**Pull request merged**